src/app/services/reducers/request-history-reducers.ts

```diff
@@ -17,6 +21,18 @@ export function history(state: any[] = [], action: IAction): any {
     case REMOVE_HISTORY_ITEM_SUCCESS:
       return state.filter(historyItem => historyItem !== action.response);
 
+    case REMOVE_HISTORY_ITEMS_BULK_SUCCESS:
+      const historyItemsToDelete = Object.values(action.response);
+      const newState: any = [];
+
+      state.forEach((historyItem: IHistoryItem) => {
```

Comment:
This is a side effect. Reducers should be pure functions. Having impure reducers affects React's rendering and can cause hard to debug errors.

This also makes deletion of history items have a time complexity of `N squared`. N, when we generate the list of keys and N in `state.forEach`.